tests/src/test/java/com/netflix/msl/userauth/UserIdTokenAuthenticationFactoryTest.java

```diff
@@ -80,7 +83,8 @@ public static void setup() throws MslEncodingException, MslCryptoException {
         ctx = new MockMslContext(EntityAuthenticationScheme.PSK, false);
         encoder = ctx.getMslEncoderFactory();
         authutils = new MockAuthenticationUtils();
-        factory = new UserIdTokenAuthenticationFactory(authutils);
+        tokenFactory = new MockTokenFactory();
+        factory = new UserIdTokenAuthenticationFactory(authutils, tokenFactory);
```

Comment:
MockMslContext.setTokenFactory() can be used to attach the MockTokenFactory to the MslContext while retaining a handle to it in order to call tokenFactory.reset().